parachains/pallets/bridge-transfer/src/lib.rs

```diff
@@ -0,0 +1,1596 @@
+				// `./Parent/Parent/...`. (It may trust Polkadot, but would
+				// Polkadot trust Kusama with its DOT?)
+
+				// Move asset to reserve account
+				T::AssetTransactor::transfer_asset(
+					&asset,
+					&origin_location,
+					&reserve_account,
+					// We aren't able to track the XCM that initiated the fee deposit, so we create a
+					// fake message hash here
+					&XcmContext::with_message_hash([0; 32]),
+				)
+					.and_then(|reserved_asset| {
+						Self::deposit_event(Event::ReserveAssetsDeposited {
+							from: origin_location,
+							to: reserve_account,
+							assets: reserved_asset.clone().into(),
+						});
+						reserved_assets.subsume_assets(reserved_asset);
+						Ok(())
+					})
+					.map_err(|e| {
+						log::error!(
+							target: LOG_TARGET,
+							"AssetTransactor failed to reserve assets from origin_location: {:?} to reserve_account: {:?} for assets: {:?}, error: {:?}",
+							origin_location,
+							reserve_account,
+							asset,
+							e
+						);
+						Error::<T>::FailedToReserve
+					})?;
+			}
+
+			// Prepare `ReserveAssetDeposited` msg to bridge to the other side.
+			// Reanchor stuff - we need to convert local asset id/MultiLocation to format that could be understood by different consensus and from their point-of-view
+			reserved_assets.reanchor(&allowed_target_location, T::UniversalLocation::get(), None);
+			let remote_destination = remote_destination
+				.reanchored(&allowed_target_location, T::UniversalLocation::get())
+				.map_err(|errored_dest| {
+				log::error!(
+					target: LOG_TARGET,
+					"Failed to reanchor remote_destination: {:?} for allowed_target_location: {:?} and universal_location: {:?}",
+					errored_dest,
+					allowed_target_location,
+					T::UniversalLocation::get()
+				);
+				Error::<T>::InvalidRemoteDestination
+			})?;
+
+			// prepare xcm message (maybe_paid + ReserveAssetDeposited stuff)
+			let mut xcm_instructions = match bridge_config.target_location_fee {
+				Some(target_location_fee) => sp_std::vec![
+					WithdrawAsset(target_location_fee.clone().into()),
+					BuyExecution { fees: target_location_fee, weight_limit: Unlimited },
+				],
+				None =>
+					sp_std::vec![UnpaidExecution { check_origin: None, weight_limit: Unlimited }],
+			};
+			xcm_instructions.extend(sp_std::vec![
+				ReserveAssetDeposited(reserved_assets.clone().into()),
+				ClearOrigin,
+				DepositAsset {
+					assets: MultiAssetFilter::from(MultiAssets::from(reserved_assets)),
+					beneficiary: remote_destination
+				}
+			]);
+
+			Self::initiate_bridge_transfer(allowed_target_location, xcm_instructions.into())
+				.map_err(Into::into)
+		}
+
+		fn initiate_bridge_transfer(dest: MultiLocation, xcm: Xcm<()>) -> Result<(), Error<T>> {
+			log::info!(
+				target: LOG_TARGET,
+				"[T::BridgeXcmSender] send to bridge, dest: {:?}, xcm: {:?}",
+				dest,
+				xcm,
+			);
+			// call bridge
+			let (message_hash, sender_cost) =
```

Comment:
@serban300 
well, I saw this kind of handling to event somewhere else :),
please, add here some todo:
```
	// TODO:check-parameter - handle sender_cost somehow?```
```

because I am not sure, if we can twice charge user, because this `send_xcm` cost should be already covered/included in weight generated by benchmark

Reply:
Added the `TODO` comment.

I'm not very familiar with benchmarks, but I'm not sure if this cost is included in the weight generated by them. For example for sending messages to sibling parachains there seems to be a configurable cost ([for statemine it is defined here for example](https://github.com/paritytech/cumulus/blob/eced0cb3d55c04c3687ff0dc62c480d098f497b0/parachains/runtimes/assets/statemine/src/lib.rs#L547)). And I'm not sure if this is included in the benchmarks.